src/main/java/com/mongodb/jdbc/MongoStatement.java

```diff
@@ -17,29 +17,36 @@
 package com.mongodb.jdbc;
 
 import com.google.common.base.Preconditions;
+import com.mongodb.MongoExecutionTimeoutException;
 import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.MongoIterable;
 import com.mongodb.jdbc.logging.AutoLoggable;
 import com.mongodb.jdbc.logging.MongoLogger;
 import java.sql.*;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
 import org.bson.BsonDocument;
 import org.bson.BsonInt32;
 import org.bson.BsonString;
 
 @AutoLoggable
-public abstract class MongoStatement<T> implements Statement {
+public class MongoStatement implements Statement {
+    private static final BsonInt32 BSON_ONE_INT_VALUE = new BsonInt32(1);
+
     // Likely, the actual mongo sql command will not
     // need a database or collection, since those
     // must be parsed from the query.
-    protected MongoDatabase currentDB;
-    protected MongoResultSet<T> resultSet;
-    protected MongoConnection conn;
+    private MongoDatabase currentDB;
+    private MongoResultSet resultSet;
+    private MongoConnection conn;
     protected boolean isClosed = false;
     protected boolean closeOnCompletion = false;
-    protected int fetchSize = 0;
-    protected int maxQuerySec = 0;
-    protected String currentDBName;
-    protected MongoLogger logger;
-    protected int statementId;
+    private int fetchSize = 0;
+    private int maxQuerySec = 0;
+    private String currentDBName;
+    private MongoLogger logger;
+    private int statementId;
+    private final BsonInt32 formatVersion = new BsonInt32(1);
```

Comment:
I remember now that I added the `formatVersion` here because it differed from the mysql version.  If we modify line 78 to be `sqlDoc.put("formatVersion", BSON_ONE_INT_VALUE);` then we can remove this and the `formatVersion` parameter from `constructQueryDocument`.

Reply:
Done! Thanks for flagging this :)